### PR TITLE
fix the unsafe float equality comparing

### PR DESCRIPTION
**Pull request by wuwenbin2006**
Comparing floating point with == or != is unsafe. With -Wfloat-equal option, gcc figures out all the unsafe expressions.

### CODE REVIEW

File: src/linalg.cpp [hunk 1]
```diff
@@ -583,7 +583,7 @@ double computeRMSE(Vector* X, Vector* Y, bool integrated, int it,
         new_source += X->getValue(i, g);
         old_source += Y->getValue(i, g);
       }
-      if (old_source != 0.0)
+    if (fabs(old_source) > FLT_EPSILON)
```
Comment (GiudGiud) [inline src/linalg.cpp]: indentation

File: src/Geometry.cpp [hunk 2]
```diff
@@ -2205,7 +2205,8 @@ void Geometry::segmentizeExtruded(Track* flattened_track,
       curr = findNextCell(&end, phi);
 
       /* Checks that segment does not have the same start and end Points */
-      if (start.getX() == end.getX() && start.getY() == end.getY())
+      if (fabs(start.getX() - end.getX()) <FLT_EPSILON &&
```
Comment (GiudGiud) [inline src/Geometry.cpp]: space before FLT_EPSILON

File: src/constants.h [hunk 3]
```diff
@@ -8,6 +8,8 @@
 #ifndef CONSTANTS_H_
 #define CONSTANTS_H_
 
+/** Threshold to determine if a float equals to 0.0 */
+#define FLT_EPSILON 1.0E-50
```
Comment (GiudGiud) [inline src/constants.h]: Did you do some testing ? How about using std::numeric_limits<T>::epsilon() ?

Comment (wuwenbin2006) [reply src/constants.h]: No, I didn't do much testing, but I found that std::numeric_limits<float>::epsilon() = 1.19209e-07 and std::numeric_limits<double>::epsilon() = 2.22045e-16, while we already have some very small 1e-20 values in the code. So I chose a smaller value 1E-50.

File: config.py [hunk 4]
```diff
@@ -156,7 +156,7 @@ class configuration:
   compiler_flags = dict()
 
   compiler_flags['gcc'] = ['-c', '-O3', '-ffast-math', '-fopenmp',
-                           '-std=c++11', '-fpic']
+                           '-std=c++11', '-fpic', '-Wfloat-equal']
```
Comment (GiudGiud) [inline config.py]: This makes installing openmoc output a lot of warnings, for Universe.cpp, TrackGenerator.cpp and Geometry.cpp. 
I would rather you delete that flag from the regular user installation process.
Alternatively, fix all the causes of warnings, but it's not really worth the time, these are not bugs.

**Pull request merged**